Added support for a JSON health endpoint (rebased)

aleksihakli opened the pull request:
Added support for JSON based health checks with tests on top of the current master branch.

aleksihakli committed:
JSON response for health checks

aleksihakli committed:
JSON healthcheck tests

aleksihakli commented:
@codingjoe I added support for the JSON responses based on the `Accept` header.

I don't know if a CBV is the best option for status rendering as we might want to customize the endpoint and add support for multiple different types of medias such as YAML etc. in the long run. Should we just use FBVs in the endpoint?

aleksihakli commented:
@codingjoe I think this is ready work-wise and good for merging from my part if it looks good to you.

codingjoe commented on health_check/views.py:
```diff
@@ -19,4 +20,11 @@ def get(self, request, *args, **kwargs):
+
+        if 'application/json' in request.META.get('HTTP_ACCEPT', ''):
+            return JsonResponse(
+                {str(p.identifier()): str(p.pretty_status()) for p in plugins},
```
I would put this part into a different method. That way users can overwrite the structure of the dict that gets serialised into a JSON

codingjoe reviewed (changes requested):
@aleksihakli I like it, just one tiny comment.
Oh and please add documentation 😉

aleksihakli committed:
Modularize JSON rendering

aleksihakli committed:
Add documentation for machine readable reports

aleksihakli commented:
@codingjoe I refactored the JSON endpoint so that it can be customized easily and added docs about the JSON output and customization to README.

I guess #31 can be closed after this as well, finally :)

codingjoe commented:
Released in 2.2.0
